op-service/sources/eth_client.go

```diff
@@ -120,11 +120,18 @@ type EthClient struct {
 	// cache BlockRef by hash
 	// common.Hash -> eth.BlockRef
 	blockRefsCache *caching.LRUCache[common.Hash, eth.BlockRef]
+
+	// any checks for if RPC response is valid
+	respChecker RPCRespCheck
+}
+
+type RPCRespCheck interface {
```

Comment:
This interface is added so we can enforce any checks on the RPC responses. Since the `EthClient` wraps both a `L1` and `L2` client, the interface methods can be implemented (or not) appropriately based on the client type.